js/site.js

```diff
@@ -3,12 +3,32 @@ class Site{
         this.domain = domain,
         this.trackers = [],
         this.score = null;
-        this.whiteListed = false;
         this.scoreFunction = scoreFunction;
+        this.setWhitelistStatusFromGlobal(domain)
     }
 
-    setWhitelisted(value){ this.whiteListed = value };
-    
+    setWhitelisted(value){ 
+        this.whiteListed = value;
+        this.setGlobalWhitelist();
+    };
+
+    setGlobalWhitelist(){
+        let globalWhitelist = settings.getSetting('whitelist');
+
+        if(!globalWhitelist){
+            globalWhitelist = {};
+        }
+
+        if(this.whiteListed){
+            globalWhitelist[this.domain] = true;
+        }
+        else {
+            delete globalWhitelist[this.domain];
+        }
+
+        settings.updateSetting('whitelist', globalWhitelist);
```

Comment:
Seems like a common pattern.

- settings has a named object
- it might not exist
- you want to set a value in the object, or use some code that knows its internal structure
- you either have a class that manages that setting object, in which case you create it around the object settings returns, or
- settings knows what type it is and creates one around the value it returns